src/amo/sagas/home.js

```diff
@@ -1,184 +1,32 @@
 /* @flow */
-import { oneLine } from 'common-tags';
-import { all, call, put, select, takeLatest } from 'redux-saga/effects';
+import { call, put, select, takeLatest } from 'redux-saga/effects';
 
-import { getCollectionAddons } from 'amo/api/collections';
-import { getHeroShelves } from 'amo/api/hero';
-import {
-  LANDING_PAGE_EXTENSION_COUNT,
-  LANDING_PAGE_THEME_COUNT,
-  MOBILE_HOME_PAGE_EXTENSION_COUNT,
-  ADDON_TYPE_EXTENSION,
-  ADDON_TYPE_STATIC_THEME,
-  RECOMMENDED,
-  SEARCH_SORT_POPULAR,
-  SEARCH_SORT_RANDOM,
-  SEARCH_SORT_TRENDING,
-} from 'amo/constants';
+import { getHomeShelves } from 'amo/api/homeShelves';
 import {
   FETCH_HOME_DATA,
   abortFetchHomeData,
   loadHomeData,
 } from 'amo/reducers/home';
-import { search as searchApi } from 'amo/api/search';
 import log from 'amo/logger';
 import { createErrorHandler, getState } from 'amo/sagas/utils';
-import type { GetCollectionAddonsParams } from 'amo/api/collections';
 import type { FetchHomeDataAction } from 'amo/reducers/home';
-import type { SearchParams } from 'amo/api/search';
 import type { Saga } from 'amo/types/sagas';
 
 export function* fetchHomeData({
-  payload: {
-    collectionsToFetch,
-    errorHandlerId,
-    includeRecommendedThemes,
-    includeTrendingExtensions,
-    isDesktopSite,
-  },
+  payload: { errorHandlerId },
 }: FetchHomeDataAction): Saga {
   const errorHandler = createErrorHandler(errorHandlerId);
 
   yield put(errorHandler.createClearingAction());
 
-  const state = yield select(getState);
-
-  const recommendedSearchFilters = {
```

Comment:
As discussed above, we're either going to have to enhance the curation tool to allow us to specify different pages for desktop vs. mobile, or we're going to have to keep the code here which retrieves just the Recommended Extensions when we are working with the mobile page. My feeling is that we should do the latter for now, and treat the changes to the curation tool as a possible follow up.